File: Day003_2025/Q003_GFG.java
import java.util.*;

class Q003_GFG {
    public long subarrayXor(int[] arr, int k) {
        // Q. Count Subarrays with given XOR
        HashMap<Integer, Integer> um = new HashMap<>();
        um.put(0, 1); // Initialize with XOR = 0
        int currXOR = 0;
        long ans = 0;

        for (int num : arr) {
            currXOR ^= num;

            if (um.containsKey(currXOR ^ k)) {
                ans += um.get(currXOR ^ k);
            }
            um.put(currXOR, um.getOrDefault(currXOR, 0) + 1);
        }

        return ans;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int tc = sc.nextInt();
        sc.nextLine(); // Consume newline after number of test cases

        while (tc-- > 0) {
            String[] input = sc.nextLine().split(" ");
            int[] arr = Arrays.stream(input).mapToInt(Integer::parseInt).toArray();
            int k = sc.nextInt();
            sc.nextLine(); // Consume newline after k

            Q003_GFG sol = new Q003_GFG();
            System.out.println(sol.subarrayXor(arr, k));
            System.out.println("~");
        }

        sc.close();
    }
}

File: Day003_2025/Q003_GFG.py
class Solution:
    def subarrayXor(self, arr, k):
        # Q. Count Subarrays with given XOR
        prefix_xor = {0: 1}  # Initialize with XOR = 0
        curr_xor = 0
        ans = 0

        for num in arr:
            curr_xor ^= num
            if curr_xor ^ k in prefix_xor:
                ans += prefix_xor[curr_xor ^ k]
            prefix_xor[curr_xor] = prefix_xor.get(curr_xor, 0) + 1

        return ans


if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        arr = list(map(int, input().split()))
        k = int(input())

        sol = Solution()
        print(sol.subarrayXor(arr, k))
        print("~")
